src/linter/ui5Types/TypeLinter.ts

```diff
@@ -112,6 +116,8 @@ export default class TypeLinter {
 		const messageDetails = this.#context.getIncludeMessageDetails();
 		const typeCheckDone = taskStart("Linting all transpiled resources");
 		for (const sourceFile of program.getSourceFiles()) {
+			this.#metadataCollector.collectControllerInfo(sourceFile); // Collect data prior to linting filter below
```

Comment:
My point regarding the additional loop over all AST nodes was not about the for-loop over the source files (which is pretty cheap). It is about the fact that within `SourceFileLinter`, we already visit all AST nodes, and in there could cover the visit of ClassDeclarations, without much overhead. For all source files that are not processed via `SourceFileLinter` (i.e. `!pathsToLint.includes(sourceFile.fileName)`), the collector should be called in here, so that all files in the project are covered.